src/main/java/seedu/address/model/reminder/ReminderManager.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.model.reminder;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.Model;
+
+/**
+ * Represents a thread that manages the reminders.
+ */
+public class ReminderManager extends Thread {
+    private static final long FREQUENCY = 1; // in minutes
+
+    private final Model model;
+    private final UniqueReminderList reminderList = UniqueReminderList.getInstance();
+    private final Object mutex;
+
+    private Logger logger = LogsCenter.getLogger(ReminderManager.class);
+
+    /**
+     * Creates a ReminderManager object.
+     *
+     * @param taskQueue
+     * @param mutex
+     */
+    public ReminderManager(Model model, Object mutex) {
+        this.model = model;
+        this.mutex = mutex;
+    }
+
+    /**
+     * Starts the ReminderManager thread.
+     */
+    public void start() {
+        super.start();
+
+        //Add all reminders to the reminderList
+        //Note: This could be done in JsonAdaptedInteractions when loading the data from the file
+        //But it is done here instead to adhere to Seperation of Concerns principle
+
+        logger.info(model.getAddressBook().getPersonList().get(1).getInteractions().toString());
+        model.getAddressBook().getPersonList().forEach(person -> {
+            person.getInteractions().forEach(interaction -> {
+                reminderList.add(person, interaction);
+            });
+        });
+
+        // Start the scheduler to wake up the ReminderManager occasionally
+        ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+        scheduler.scheduleAtFixedRate(() -> {
+            synchronized (mutex) {
+                mutex.notifyAll();
+            }
+        }, 0, FREQUENCY, TimeUnit.MINUTES); // Wakes up every 10 minutes
+    }
+
+    @Override
+    public void run() {
```

Comment:
I don't exactly know how threading works in Java but do ensure that the thread is destroyed after the application closes in case you haven't yet